### ISSUE TITLE
[49.19.1] Flak rounds do not roll cluster when used during a rapid fire AutoCannon attack

### ISSUE THREAD

**Issue by RaozSpaz** [MegaMek/megamek#5829]
### Environment
49.19.1
Windows
Java 11

### Description
I can obtain screenshots if necessary but I've seen it enough times due to my turrets being added into MM that I know it occurs. At least in 49.19.1 None the less
https://bg.battletech.com/forums/index.php?topic=85660.0
In this rules question I asked if munitions that were supposed to use the cluster table needed to roll cluster again after rolling cluster for number of shots that hit. With the answer being yes, this is very much not the methodology used within MM. Most likely because it is a VERY unique situation that most of us never really thought about.

If anyone wants me to obtain a screenshot just poke me here or on the Discord. I can do it fairly quick.

### Files
N/A

**Comment by Saklad5** [comment]
I can confirm this bug still exists in 0.50.0.
<img width="1002" alt="image" src="https://github.com/user-attachments/assets/b827e829-7338-4f22-864e-d4228c631996">
Ow.

**Comment by RaozSpaz** [comment]
Issue still exists in 50.06

![Image](https://github.com/user-attachments/assets/f127d99c-b9ec-47b7-b0b2-abc96bb1d987)

**Comment by Sleet01** [comment]
Yeah, just didn't have time to get to it.  IIRC we use the base weapon damage instead of the loaded ammo when handling rapid fire shots; not sure what the best solution will be but it may involve spawning N auto-hitting attacks with the current ammo where N is the number of loaded ammo shots remaining or the number of hits rolled, whichever is less.
